src/inference/src/threading/ie_istreams_executor.cpp

```diff
@@ -46,6 +46,53 @@ int IStreamsExecutor::Config::GetDefaultNumStreams() {
         return 1;
 }
 
+int IStreamsExecutor::Config::GetHybridDefaultNumStreams(const Config& config) {
+    const int num_phy_cores = getNumberOfCPUCores();
+    const int num_big_cores = getNumberOfCPUCores(true);
+    const int num_small_cores = num_phy_cores - num_big_cores;
+
+    // bare minimum of streams (that evenly divides available number of core)
+    if (0 == num_big_cores % 4) {
+        config._big_core_streams = std::max(4, num_big_cores / 4);
+    } else if (0 == num_big_cores % 5) {
+        config._big_core_streams = std::max(5, num_big_cores / 5);
+    } else if (0 == num_big_cores % 3) {
+        config._big_core_streams = std::max(3, num_big_cores / 3);
+    } else {  // if user disables some cores say in BIOS, so we got weird #cores which is not easy to divide
+        config._big_core_streams = 1;
+    }
+
+    config._threads_per_stream_big = num_big_cores / config._big_core_streams;
+    config._threads_per_stream_small = config._threads_per_stream_big * 2;
+    if (num_small_cores < config._threads_per_stream_small) {
+        config._small_core_streams = 1;
+        config._threads_per_stream_small = num_small_cores;
+        config._threads_per_stream_big = config._threads_per_stream_small / 2;
+        config._big_core_streams = std::floor(num_big_cores / config._threads_per_stream_big);
+    } else {
+        config._small_core_streams = num_small_cores / config._threads_per_stream_small;
+    }
+    config._smallCoreOffset = num_big_cores * 2;
+    return config._big_core_streams + config._small_core_streams;
+}
+
+int IStreamsExecutor::Config::GetHybridNonDefaultNumStreams(const Config& config, const int stream_mode) {
+    const int num_phy_cores = getNumberOfCPUCores();
+    const int num_big_cores = getNumberOfCPUCores(true);
+    const int num_small_cores = num_phy_cores - num_big_cores;
+
+    if (stream_mode == AGGRESSIVE) {
```

Comment:
Do we need check DEFAULT mode? If no, DEFAULT mode will go to `else branch`.

Reply:
Add parameter checking.